### DISCUSSION TITLE
Error on start up.

### DISCUSSION THREAD

**Discussion by zephod77** [nvim-tree/nvim-tree.lua#3023] [Q&A] [answered]
I have started getting this error when I start nvim. 
This config used to work but maybe an update broke it? Lazy is at release 11.16.0 (#1835). Not sure about nvim-tree but Lazy says it is up to date.

> Failed to run `config` for nvim-tree.lua
>                                                                                                                                           
> ...nvim/lazy/nvim-tree.lua/lua/nvim-tree/node/directory.lua:14: attempt to index local 'Node' (a boolean value)                           
>                                                                                                                                           
> # stacktrace:                                                                                                                             
>   - .config/nvim/lua/plugins/nvim-tree.lua:32 _in_ **config**                                                                             
>   - .config/nvim/init.lua:48 

What is this error trying to tell me other than some thing is wrong?

My config in lua/plugins/nvim-tree.lua is very simple:
```
vim.g.have_nerd_font = true

vim.keymap.set('n', '<leader>t', ':NvimTreeToggle<CR>', { noremap = true, silent = true })
vim.keymap.set('n', '<leader>m', ':NvimTreeFocus<CR>', { noremap = true, silent = true })

local opts = {
    filters = {
        dotfiles = false,
    },
}

return {
    "nvim-tree/nvim-tree.lua",
    lazy = false,
    opts = opts,
}
```

**Comment by alex-courtis** [comment, marked as answer]
It looks like it failed whilst loading - it couldn't find the `nvim-tree.node` module.
What is `.config/nvim/lua/plugins/nvim-tree.lua`? It doesn't look like part of the plugin. Is that a file created by lazy?

Please try a [clean room replication](https://github.com/nvim-tree/nvim-tree.lua/wiki/Installation#lazy-loading). This will indicate whether it's an issue with nvim-tree or with lazy, which is [known to be problematic](https://github.com/nvim-tree/nvim-tree.lua/wiki/Installation#lazy-loading).

**Reply by zephod77** [reply]
I load Lazy in my .config/nvim/init.lua file and then tell Lazy to load all the plugins in my .config/nvim/lua/plugins directory, nvim-tree.lua being one of them. Interestingly, I have a very similar setup on another computer and I do not have this problem. The differences are that I have nvim-tree setup to change the width of the tree pane on a key stroke and the version of Lazy is 11.14.2.

**Reply by zephod77** [reply]
So I have made the nvim-tree configuration and the Lazy version identical on both machines. I still get the error on one of them.

**Reply by gegoune** [reply]
Report it to lazy then, if you think it's an issue with nvim-tree please follow Alex's instruction above.

**Reply by zephod77** [reply]
I created a new user and copied my .config/lua files over to the new user one by one, starting nvim each time and checking for the error.
It never showed up.

Then I moved my .config/nvim to a backup location and copied the "working" config from the new user back to the user that experiences the problem. I still get the same error.

The conclusion I take from this is that there is nothing wrong with nvim-tree or Lazy but there is something strange in the environment for this user.

Is it safe to delete .local/share/nvim and/or .local/state/nvim?

**Reply by alex-courtis** [reply]
> `.local/state/nvim`

Absolutely. You'll lose undo and shada etc., but that's not a big issue.

> `.local/share/nvim`

Just `.local/share/nvim/site` should be enough. It's likely there are conflicting versions of plugins from various package managers.

**Reply by zephod77** [reply]
OK, before I saw your post, I had deleted both directories and that fixed the problem. I only have one package manager unless you count Mason as a second one. So  I'm guessing an update messed up the state somehow.
Thanks for the help.
